src/components/EnvironmentOverride/SecretOverrides.tsx

```diff
@@ -535,13 +556,16 @@ export function OverrideSecretForm({ name, appChartRef, toggleCollapse }) {
             setSecretDataYaml(yaml)
         }
         try {
-            if (!yaml || !yaml.length) {
-                setEsoData([])
+            let json = YAML.parse(yaml)
+            if (!json) {
```

Comment:
Make sure it's not returning a blank object as a blank object is also a valid object.

Reply:
Checked